quantum/pointing_device/pointing_device_drivers.c

```diff
@@ -115,6 +115,137 @@ const pointing_device_driver_t pointing_device_driver = {
+    i2c_init();
+    azoteq_iqs5xx_wake();
+    if (azoteq_iqs5xx_get_product() != UNKNOWN) {
+        azoteq_iqs5xx_setup_resolution();
+        azoteq_iqs5xx_init_status = azoteq_iqs5xx_set_report_rate(AZOTEQ_IQS5XX_REPORT_RATE, ACTIVE, false);
+        azoteq_iqs5xx_init_status |= azoteq_iqs5xx_set_event_mode(AZOTEQ_IQS5XX_EVENT_MODE, false);
+        azoteq_iqs5xx_init_status |= azoteq_iqs5xx_set_gesture_config(true);
+        wait_ms(AZOTEQ_IQS5XX_REPORT_RATE + 1);
+    }
+};
+
+report_mouse_t azoteq_iqs5xx_rotate(report_mouse_t mouse_report) {
+    report_mouse_t old_report = mouse_report;
+
+#    if defined(POINTING_DEVICE_ROTATION_90) || defined(POINTING_DEVICE_ROTATION_180) || defined(POINTING_DEVICE_ROTATION_270)
+#        if defined(POINTING_DEVICE_ROTATION_90)
+    mouse_report.h            = old_report.v;
+    mouse_report.v            = -old_report.h;
+#        elif defined(POINTING_DEVICE_ROTATION_180)
+    mouse_report.h                         = -old_report.h;
+    mouse_report.v                         = -old_report.v;
+#        elif defined(POINTING_DEVICE_ROTATION_270)
+    mouse_report.h                         = -old_report.v;
+    mouse_report.v                         = old_report.h;
+#        endif
+#    endif
+    return mouse_report;
+}
+
+azoteq_iqs5xx_base_data_t azoteq_iqs5xx_rotate_swipes(azoteq_iqs5xx_base_data_t base_data) {
+    azoteq_iqs5xx_base_data_t old_base_data = base_data;
+
+#    if defined(POINTING_DEVICE_ROTATION_90) || defined(POINTING_DEVICE_ROTATION_180) || defined(POINTING_DEVICE_ROTATION_270)
+#        if defined(POINTING_DEVICE_ROTATION_90)
+    base_data.gesture_events_0.swipe_x_neg  = old_base_data.gesture_events_0.swipe_y_neg;
+    base_data.gesture_events_0.swipe_x_pos  = old_base_data.gesture_events_0.swipe_y_pos;
+    base_data.gesture_events_0.swipe_y_neg  = old_base_data.gesture_events_0.swipe_x_pos;
+    base_data.gesture_events_0.swipe_y_pos  = old_base_data.gesture_events_0.swipe_x_neg;
+#        elif defined(POINTING_DEVICE_ROTATION_180)
+    base_data.gesture_events_0.swipe_x_neg = old_base_data.gesture_events_0.swipe_x_pos;
+    base_data.gesture_events_0.swipe_x_pos = old_base_data.gesture_events_0.swipe_x_neg;
+    base_data.gesture_events_0.swipe_y_neg = old_base_data.gesture_events_0.swipe_y_pos;
+    base_data.gesture_events_0.swipe_y_pos = old_base_data.gesture_events_0.swipe_y_neg;
+#        elif defined(POINTING_DEVICE_ROTATION_270)
+    base_data.gesture_events_0.swipe_x_neg = old_base_data.gesture_events_0.swipe_y_pos;
+    base_data.gesture_events_0.swipe_x_pos = old_base_data.gesture_events_0.swipe_y_neg;
+    base_data.gesture_events_0.swipe_y_neg = old_base_data.gesture_events_0.swipe_x_neg;
+    base_data.gesture_events_0.swipe_y_pos = old_base_data.gesture_events_0.swipe_x_pos;
+#        endif
+#    endif
+    return base_data;
+}
+
+report_mouse_t azoteq_iqs5xx_get_report(report_mouse_t mouse_report) {
+    static uint8_t previous_button_state = 0;
```

Comment:
I am getting an unused variable error when building with `POINTING_DEVICE_MOTION_PIN` undefined.

Reply:
Thanks, should be shorted in 2973a06